[CI] JobUpdateTests testMergeWithJob_GivenRandomUpdates_AssertImmutability failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/4fic72iky3xoi/tests/:x-pack:plugin:core:test/org.elasticsearch.xpack.core.ml.job.config.JobUpdateTests/testMergeWithJob_GivenRandomUpdates_AssertImmutability

**Reproduction line:**
`./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ml.job.config.JobUpdateTests.testMergeWithJob_GivenRandomUpdates_AssertImmutability" -Dtests.seed=4AE94F4C473FF8A5 -Dtests.locale=en-MT -Dtests.timezone=Asia/Ho_Chi_Minh -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.core.ml.job.config.JobUpdateTests&tests.test=testMergeWithJob_GivenRandomUpdates_AssertImmutability

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: not <{"job_id":"gskdiishqq","job_type":"anomaly_detector","job_version":"8.0.0","create_time":7288975506043542405,"model_snapshot_id":"qxodsOpPuc","deleting":true,"model_snapshot_min_version":"8.0.0","description":"lQlQiVCBvN","analysis_config":{"bucket_span":"5m","categorization_field_name":"Fxfunaklxw","categorization_analyzer":{"char_filter":[{"pattern":"pqIaEjwpoeiMXZgpxdVueWDULjEKPQAUVKbeRzcc","type":"pattern_replace"},{"pattern":"zCVPgCcjaHfZwEpOSxALpqNnehgfLSDuPGxvMzqf","type":"pattern_replace"},{"pattern":"easzoCFBqbQeCxHqWFMgghdkSdyOXfXSUCDofvkq","type":"pattern_replace"}],"tokenizer":{"pattern":"pQeqCUxLyu","type":"pattern"}},"per_partition_categorization":{"enabled":false},"detectors":[{"detector_description":"count by mlcategory partitionfield=part","function":"count","by_field_name":"mlcategory","partition_field_name":"part","detector_index":0},{"detector_description":"count by mlcategory partitionfield=part","function":"count","by_field_name":"mlcategory","partition_field_name":"part","detector_index":1}],"influencers":["pQMaTMuZoA"],"multivariate_by_fields":true},"analysis_limits":{"model_memory_limit":"810819mb","categorization_examples_limit":8844450584755514242},"data_description":{"time_field":"time","time_format":"epoch_ms"},"model_plot_config":{"enabled":false,"terms":"SnMAmMdVOgJEqIpaqN","annotations_enabled":false},"model_snapshot_retention_days":2130746329194365624,"results_index_name":"shared","allow_lazy_open":true,"blocked":{"reason":"delete"}}>
     but: was <{"job_id":"gskdiishqq","job_type":"anomaly_detector","job_version":"8.0.0","create_time":7288975506043542405,"model_snapshot_id":"qxodsOpPuc","deleting":true,"model_snapshot_min_version":"8.0.0","description":"lQlQiVCBvN","analysis_config":{"bucket_span":"5m","categorization_field_name":"Fxfunaklxw","categorization_analyzer":{"char_filter":[{"pattern":"pqIaEjwpoeiMXZgpxdVueWDULjEKPQAUVKbeRzcc","type":"pattern_replace"},{"pattern":"zCVPgCcjaHfZwEpOSxALpqNnehgfLSDuPGxvMzqf","type":"pattern_replace"},{"pattern":"easzoCFBqbQeCxHqWFMgghdkSdyOXfXSUCDofvkq","type":"pattern_replace"}],"tokenizer":{"pattern":"pQeqCUxLyu","type":"pattern"}},"per_partition_categorization":{"enabled":false},"detectors":[{"detector_description":"count by mlcategory partitionfield=part","function":"count","by_field_name":"mlcategory","partition_field_name":"part","detector_index":0},{"detector_description":"count by mlcategory partitionfield=part","function":"count","by_field_name":"mlcategory","partition_field_name":"part","detector_index":1}],"influencers":["pQMaTMuZoA"],"multivariate_by_fields":true},"analysis_limits":{"model_memory_limit":"810819mb","categorization_examples_limit":8844450584755514242},"data_description":{"time_field":"time","time_format":"epoch_ms"},"model_plot_config":{"enabled":false,"terms":"SnMAmMdVOgJEqIpaqN","annotations_enabled":false},"model_snapshot_retention_days":2130746329194365624,"results_index_name":"shared","allow_lazy_open":true,"blocked":{"reason":"delete"}}>

  at __randomizedtesting.SeedInfo.seed([4AE94F4C473FF8A5:3865E05D896DF141]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.core.ml.job.config.JobUpdateTests.testMergeWithJob_GivenRandomUpdates_AssertImmutability(JobUpdateTests.java:323)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdateTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdateTests.java
@@ -130,7 +130,7 @@ public class JobUpdateTests extends AbstractSerializingTestCase<JobUpdate> {
         if (randomBoolean()) {
             update.setAllowLazyOpen(randomBoolean());
         }
-        if (useInternalParser && randomBoolean()) {
+        if (useInternalParser && randomBoolean() && job.isDeleting() == false) {
             update.setBlocked(BlockedTests.createRandom());
         }
         if (randomBoolean() && job != null) {

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdateTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdateTests.java
@@ -130,7 +130,7 @@ public class JobUpdateTests extends AbstractSerializingTestCase<JobUpdate> {
         if (randomBoolean()) {
             update.setAllowLazyOpen(randomBoolean());
         }
-        if (useInternalParser && randomBoolean() && job.isDeleting() == false) {
+        if (useInternalParser && randomBoolean() && (job == null || job.isDeleting() == false)) {
             update.setBlocked(BlockedTests.createRandom());
         }
         if (randomBoolean() && job != null) {

